deepchem/models/torch_models/grover.py

```diff
@@ -92,3 +98,312 @@ def forward(self, graph_batch: List[torch.Tensor]):
                 "atom_from_bond": output[1][0],
                 "bond_from_bond": output[1][1]
             }
+
+
+class _BondVocabPredictionLayer(nn.Module):
+    """Layer for learning contextual information for bonds.
+
+    It takes a bond embedding, predict its context by formulating a multi-class classification problem.
+    """
+
+    def __init__(self, vocab_size, hidden_size=128):
+        super(_BondVocabPredictionLayer, self).__init__()
+        self.linear = nn.Linear(hidden_size, vocab_size)
+        self.linear_rev = nn.Linear(hidden_size, vocab_size)
+        self.logsoftmax = nn.LogSoftmax(dim=1)
+
+    def forward(self, embeddings):
+        """
+        Parameters
+        ----------
+        embeddings: torch.Tensor
+            bond embeddings of shape num_bond X fea_dim
```

Comment:
Can you use format `(num_bond, feat_dim)`? Where are `num_bond` and `feat_dim` defined? Can you add a definition to the docstring?

Reply:
Fixed in #3277